boards/v2/qemu/qemu_arc/Kconfig

```diff
@@ -0,0 +1,23 @@
+# Copyright (c) 2020 Synopsys, Inc. All rights reserved.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_QEMU_ARC
+	select SOC_QEMU_ARC
+	select QEMU_TARGET
+	select HAS_COVERAGE_SUPPORT
+
+config BOARD_QEMU_ARC_QEMU_EM4
```

Comment:
I'd prefer sticking to "em" for this one, instead of "em4". This is a 'generic' board for the family, e.g. covering em6 as well. And, there are few more places (e.g. support/mdb_em.args) that have the same name.
Let's not change names as part of this restructuring. If for some reason it needs to be changed, then have a separate PR for that.

Reply:
This is the CPU, the CPU being em4, it cannot be em because there is no such soc

Reply:
The original SoC was also called EM, not sure I understand the issue with sticking with EM and HS instead of renaming all of these (globally: soc/board/docs).

Reply:
Why were two soc added with invalid names originally? em and hs are not SoCs, they are soc families, em4 and hs3x are the soc names according to https://www.synopsys.com/designware-ip/processor-solutions/arc-classic-processors/arc-hs-family.html and https://www.synopsys.com/designware-ip/processor-solutions/arc-classic-processors/arc-em-family.html

Reply:
ARC is highly configurable, instead of adding a SoC for all possible configurations we made an abstraction.
nsim_em e.g. covers both EM4 (no caches) and EM6 (with caches).
em4 and hs3x are not SoC names. These are product names for Synopsys ARC CPU IP cores (which can be configured and integrated on a SoC).

Reply:
With hwmv2, there is not the level of configurability as was present with hwmv1, so for example with hwmv1 you could configure for an stm32 board which has a soc, but you could then change the soc part number, in hwmv2 this is no longer possible, you configure and build for a specific soc, if you want a different part number, you need to alter the board files to specify that soc (if it isn't already one of the possible soc options) but you cannot change the configuration once configured. So with the em and hs socs, is that what you mean by configurability or is it something else?

Reply:
No, I mean silicon IP core configurability. That is several steps before board/soc configurability.
Let's separate hwmv2 porting and other possible improvements like changing names. My goal is to minimize changes for existing users. If you feel strongly about the names, then let's have a separate issue/PR/discussion about that.